system/src/control/mesh.cpp

```diff
@@ -911,6 +944,793 @@ int scanNetworks(ctrl_request* req) {
+                            pbChild.timeout = entry.GetTimeout();
+                            pbChild.child_id = entry.GetChildId();
+                            pbChild.mode = entry.GetMode();
+                            if (!pb_encode_tag_for_field(strm, field)) {
+                                return false;
+                            }
+                            if (!pb_encode_submessage(strm, PB(DiagnosticInfo_ChildTable_ChildEntry_fields), &pbChild)) {
+                                return false;
+                            }
+                        }
+                        return true;
+                    };
+
+                    if (diagResult->pbReq->flags & PB(GetNetworkDiagnosticsRequest_Flags_QUERY_CHILDREN)) {
+                        // Make child RLOC set
+                        for (unsigned i = 0; i < childTable->GetNumEntries(); i++) {
+                            const auto& entry = childTable->GetEntry(i);
+                            uint16_t childRloc = rloc | entry.GetChildId();
+                            if (!diagResult->children.contains(childRloc)) {
+                                diagResult->children.append(childRloc);
+                            }
+                        }
+                    }
+                }
+                break;
+            }
+            case NetworkDiagnosticTlv::kChannelPages: {
+                auto channelPagesTlv = (ChannelPagesTlv*)(tlv);
+                if (channelPagesTlv->IsValid()) {
+                    // Only 1 channel page supported in OpenThread at the moment
+                    pbRep.channel_pages.size = 1;
+                    pbRep.channel_pages.bytes[0] = *channelPagesTlv->GetChannelPages();
+                }
+                break;
+            }
+            case NetworkDiagnosticTlv::kTypeList: {
+                // Should not be returned
+                break;
+            }
+            case NetworkDiagnosticTlv::kMaxChildTimeout: {
+                auto timeoutTlv = static_cast<const MaxChildTimeoutTlv*>(tlv);
+                if (timeoutTlv->IsValid()) {
+                    pbRep.max_child_timeout = timeoutTlv->GetTimeout();
+                }
+                break;
+            }
+        }
+    }
+
+    if (pbRep.rloc == 0x0000) {
+        // Even if we did not request RLOC TLV, we can still deduce it from the source address
+        pbRep.rloc = rloc;
+    }
+
+    if (diagResult->pbReq->flags & PB(GetNetworkDiagnosticsRequest_Flags_RESOLVE_DEVICE_ID)) {
+        for (const auto& entry : diagResult->deviceIdMapping) {
+            if (entry.resolved && entry.rloc == pbRep.rloc) {
+                pbRep.device_id.size = HAL_DEVICE_ID_SIZE;
+                memcpy(pbRep.device_id.bytes, entry.deviceId, HAL_DEVICE_ID_SIZE);
+                break;
+            }
+        }
+    }
+
+    auto r = appendReplySubmessage(diagResult->req, diagResult->written, PB(GetNetworkDiagnosticsReply_fields),
```

Comment:
Let's make the intent clear and pass `&PB(GetNetworkDiagnosticsReply_fields[0])` as the `field` argument, or add a comment here :)